criu/cr-restore.c

```diff
@@ -3360,16 +3354,35 @@ static bool groups_match(gid_t *groups, int n_groups)
 	return ret;
 }
 
+static int copy_caps(u32 *out_caps, u32 *in_caps, int n_words)
+{
+	int last_cap = min(kdat.last_cap, 32 * CR_CAP_SIZE - 1);
+	int i;
+
+	for (i = last_cap + 1; i < 32 * n_words; ++i) {
+		if (~in_caps[i / 32] & (1 << (i % 32)))
+			continue;
+
+		if (!opts.unprivileged) {
+			pr_err("Can't restore capability %d > min(kernel=%d, criu=%d)", i, kdat.last_cap, 32 * CR_CAP_SIZE - 1);
+			return -EPERM;
+		}
+		pr_warn("Can't restore capability %d > min(kernel=%d, criu=%d)", i, kdat.last_cap, 32 * CR_CAP_SIZE - 1);
```

Comment:
The commit message (and comment in this case) has to explain why it must work this way. Right now, it is unclear why it doesn't fail if unprivileged is set.